### DISCUSSION TITLE
How to change the scr attribute of <Image /> component on useEffect

### DISCUSSION THREAD

**Discussion by amitleuva1987** [vercel/next.js#40014] [Help] [answered]
I want to change the src attribute of the <Image /> component to change the image displaying the page.

Actually, It's a profile page. I want to change the src of the <Image /> component when a user change the image.  I am trying this using useEffect but image src is not getting updated.   

Is there a way to change the src without refreshing the page?

**Comment by thanhtutzaw** [comment]
when profile changed , setNewProfle('new url') in useEffect and also add profile dependency to listen data changes .

in src `NewProfile ? 'new profile' : 'original profile'`

**Reply by amitleuva1987** [reply]
Thanks for the help. It works.

**Comment by bk973** [comment, marked as answer]
set up state to track [src] attribute and then update the state [setSrc] whenever a change event (When a user changes the image) happens. React will rerender the component with changes if [ src ] is changed.

**Reply by amitleuva1987** [reply]
It worked for me

**Comment by amitleuva1987** [comment]
Thanks Benjamin, Your suggestion works.

Why am I not able to see this thread over github? Has it been deleted?

I have to write a reply from my gmail.

Anyways, thank you both of you guys.

**Comment by amitleuva1987** [comment]
I have one more issue.

I use a url to retrieve a profile picture in the next js front end. So, the
"src" attribute would never change, it will always be the same url.

Is there a way to deal with it?

 Regards,
Amit
